src/sudo/pipeline/list.rs

```diff
@@ -0,0 +1,176 @@
+                        scope,
+                        context.current_user.uid,
+                        &context.current_user.name,
+                        prior_validity,
+                    );
+
+                    self.authenticator.init(&context)?;
+                    if auth_status.must_authenticate() {
+                        self.authenticator
+                            .authenticate(context.non_interactive, allowed_attempts)?;
+                        if let (Some(record_file), Some(scope)) =
+                            (&mut auth_status.record_file, scope)
+                        {
+                            match record_file.create(scope, context.current_user.uid) {
+                                Ok(_) => (),
+                                Err(e) => {
+                                    auth_warn!(
+                                        "Could not update session record file with new record: {e}"
+                                    );
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+
+            Authorization::Forbidden => {
+                if context.current_user.uid == 0 {
+                    if original_command.is_some() {
+                        return Err(Error::Silent);
+                    }
+
+                    println!(
+                        "User {} is not allowed to run sudo on {}.",
+                        other_user.as_ref().unwrap_or(&context.current_user).name,
+                        context.hostname
+                    );
+                    return Ok(());
+                } else {
+                    let command = if other_user.is_none() {
+                        "sudo".to_string()
+                    } else if original_command.is_none() {
+                        "list".to_string()
+                    } else {
+                        format!("list{}", context.command.command.display())
+                    };
+
+                    return Err(Error::NotAllowed {
+                        username: context.current_user.name,
+                        command,
+                        hostname: context.hostname,
+                        other_user: other_user.map(|user| user.name),
+                    });
+                }
+            }
+        }
+
+        if let Some(other_user) = &other_user {
+            let list_request = ListRequest {
+                target_user: &context.target_user,
+                target_group: &context.target_group,
+            };
+            let judgement = pre.check_list_permission(other_user, &context.hostname, list_request);
+
+            if let Authorization::Forbidden = judgement.authorization() {
+                let command = if original_command.is_none() {
+                    "list".to_string()
+                } else {
+                    format!("list{}", context.command.command.display())
```

Comment:
I don't really understand what the point is of emulating this sudo behavior? It just seems very weird to me.

Reply:
it was simpler that coming up with a different format. "simpler" as in the compliance tests didn't need any change. diagnostics are user facing / not machine readable so they can be changed at any time without that being consider a (semver) breaking change.

Reply:
Note that `list` is a pretty recent addition to sudo (I think it was first part of a sudo release after the kick-off of this project), so I feel we have a large degree in freedom in how far to support it (even decide to not support it if we want), based on the criteria we discussed in Berlin (since it's not in wide common usage at the moment).

Anyway, the way original sudo displays this error message feels like it may simply be a mistake in some error formatting function in original sudo that we should communicate to @millert.

For us, if we go for the "it's simpler for now" solution in this PR, we should make an issue to address this error message so we don't forget to fix it later.

Reply:
Can you elaborate on the sudo error message you are talking about?